crates/store/src/genesis/config/mod.rs

```diff
@@ -0,0 +1,286 @@
+        Ok(me)
+    }
+
+    /// Convert the in memory representation into the new genesis state
+    ///
+    /// Notice: Generates keys and returns the secret keys, hence this is not sane to be used
+    /// for production environments. There, you want to generate the keys externally.
+    pub fn into_state(self) -> Result<(GenesisState, TestSecrets), Error> {
+        let version = self.version;
+        let timestamp = self.timestamp;
+        let repr_accounts = self.account;
+        let repr_faucets = self.faucet;
+        let mut all_accounts = Vec::new();
+
+        // Every asset sitting in a wallet, has to reference a faucet for that asset
+        let mut faucets = HashMap::<String, AccountId>::new();
+
+        // Collect the generated secret keys for the test, so one can interact with those
+        // accounts/sign transactions
+        let mut secrets = Vec::<(AccountType, AccountId, SecretKey)>::new();
+
+        // first setup all the faucets
+        for FaucetRepr {
+            symbol,
+            decimals,
+            max_supply,
+            storage_mode,
+            fungible,
+        } in repr_faucets
+        {
+            let mut rng = ChaCha20Rng::from_seed(rand::random());
+            let secret = SecretKey::with_rng(&mut get_rpo_random_coin(&mut rng));
+            let auth = RpoFalcon512::new(secret.public_key());
+            let init_seed: [u8; 32] = rng.random();
+
+            let token_symbol = TokenSymbol::new(&symbol)?;
+            let max_supply = Felt::try_from(max_supply).map_err(|_| {
+                Error::MaxSupplyExceedsFieldModulus { max_supply, modulus: Felt::MODULUS }
+            })?;
+
+            let account_type = if fungible {
+                AccountType::FungibleFaucet
+            } else {
+                AccountType::NonFungibleFaucet
+            };
+
+            if !fungible {
+                return Err(Error::UnsupportedValue {
+                    key: "fungible",
+                    value: false.to_string(),
+                    message: "Not supported just yet".to_owned(),
+                });
+            }
+
+            let component = BasicFungibleFaucet::new(token_symbol, decimals, max_supply)?;
+
+            let account_storage_mode = storage_mode.into();
+
+            // similar to `fn create_basic_fungible_faucet`, but we need to cover more cases
+            let (faucet_account, _faucet_account_seed) = AccountBuilder::new(init_seed)
+                .account_type(account_type)
+                .storage_mode(account_storage_mode)
+                .with_component(auth)
+                .with_component(component)
+                .build()?;
+
+            faucets.insert(symbol, faucet_account.id());
+
+            secrets.push((account_type, faucet_account.id(), secret));
+
+            all_accounts.push(faucet_account);
+        }
+
+        // then setup all wallet accounts, which reference the faucet's for their provided assets
```

Comment:
nit: I think almost all our comments begin with capital letters. And many end with a full stop. Not sure if we have discussed comment style explicitly in the past.

Reply:
We haven't had a discussion; though I think using proper sentences makes sense.

Reply:
The first part of the sentence was further above, but got lost.

Sidequest: What's your take on punctuation? Titles (=first line in a comment) with `.` or without?

Reply:
If they're actually titles then without `.`, if you just mean a singular sentence then use a `.`. Mostly because extending the comment then results in more of a diff; same way a postfix `,` should always be required/allowed by a programming language (shame on you `json` for doing the wrong thing).

I won't really cry about any of these though :)